From b115930d716defc3a7daa2bc2ae2465d864b7114 Mon Sep 17 00:00:00 2001
From: Daniel Lezcano <daniel.lezcano@oss.qualcomm.com>
Date: Tue, 11 Aug 2026 11:47:47 +0200
Subject: [PATCH 14711/15333] thermal/drivers/armada: Fix missing bitfields
 include
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add the missing include leading to the error:

error: implicit declaration of function ‘FIELD_GET’ [-Werror=implicit-function-declaration]
  184 |                 if (FIELD_GET(MON_FAULT_STATUS_MASK, val) == MON_FAULT_LVL1_UPR)
      |                     ^~~~~~~~~
cc1: all warnings being treated as errors

Fixes: cbe31d5ce498 ("thermal/drivers/armada: Use bitfield and bitmask macros")
Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202608082242.drjXuzsN-lkp@intel.com/
Signed-off-by: Daniel Lezcano <daniel.lezcano@oss.qualcomm.com>
Reviewed-by: Miquel Raynal <miquel.raynal@bootlin.com>
Link: https://patch.msgid.link/20260811094747.2940616-1-daniel.lezcano@kernel.org
---
 drivers/thermal/armada_thermal.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/thermal/armada_thermal.c b/drivers/thermal/armada_thermal.c
index 912c68a44bdc..ff63b4f2b977 100644
--- a/drivers/thermal/armada_thermal.c
+++ b/drivers/thermal/armada_thermal.c
@@ -4,6 +4,7 @@
  *
  * Copyright (C) 2013 Marvell
  */
+#include <linux/bitfield.h>
 #include <linux/device.h>
 #include <linux/err.h>
 #include <linux/io.h>
-- 
2.55.0

